docs/source/tutorials/06_using_your_own_models.rst

```diff
@@ -0,0 +1,578 @@
+
+.. include:: /include/substitution.rst
+
+PeekingDuck offers pre-trained :mod:`model` nodes that can be used to tackle a wide variety of
+problems, but you may need to train your own model on a custom dataset sometimes. This tutorial
+will show you how to package your model into a custom :mod:`model` node, and use it with
+PeekingDuck. We will be tackling a manufacturing use case here - classifying images of steel
+castings into "defective" or "normal" classes.
+
+Casting is a manufacturing process in which a material such as metal in liquid form is poured into
+a mold and allowed to solidify. The solidified result is also called a casting. Sometimes,
+defective castings are produced, and quality assurance departments are responsible for preventing
+defective pieces from being used downstream. As inspections are usually done manually, this adds
+a significant amount of time and cost, and thus there is an incentive to automate this process.
+
+The images of castings used in this tutorial are the front faces of steel `pump impellers 
+<https://en.wikipedia.org/wiki/Impeller>`_. From the comparison below, it can be seen that the
+defective casting has a rough, uneven edges while the normal casting has smooth edges.
+
+   .. figure:: /assets/tutorials/normal_vs_defective.png
+      :width: 416
+      :alt: Normal casting compared to defective casting
+
+      Normal Casting Compared to Defective Casting
+
+
+.. _tutorial_model_training:
+
+Model Training
+==============
+
+PeekingDuck is designed for model *inference* rather than model *training*. This optional section
+shows how a simple Convolutional Neural Network (CNN) model can be trained separately from the
+PeekingDuck framework. If you have already trained your own model, the
+:ref:`following section <tutorial_using_your_trained_model_with_peekingduck>` describes how you can
+convert it to a custom model node, and use PeekingDuck for inference.
+
+
+
+Setting Up
+----------
+
+Create the following project folder:
+
+.. admonition:: Terminal Session
+
+   | \ :blue:`[~user]` \ > \ :green:`mkdir castings_project` \
+   | \ :blue:`[~user]` \ > \ :green:`cd castings_project` \
+
+Within the ``castings_project`` folder:
+
+* Download the `castings dataset <https://storage.googleapis.com/peekingduck/data/castings_data.zip>`_
+  and unzip the file.
+* Create an empty Python script named ``train_classifier.py``.
+
+.. note::
+
+   The castings dataset used in this example is modified from the original dataset from
+   `Kaggle <https://www.kaggle.com/datasets/ravirajsinh45/real-life-industrial-dataset-of-casting-product?resource=download&select=casting_data>`_.
+
+
+You should have the following directory structure at this point:
+
+.. parsed-literal::
+
+   \ :blue:`castings_project/` \ |Blank|
+   ├── train_classifier.py
+   └── \ :blue:`castings_data/` \ |Blank|
+         ├── \ :blue:`train/` \ |Blank|
+         ├── \ :blue:`validation/` \ |Blank|
+         └── \ :blue:`test/` \ |Blank|
```

Comment:
I got a different directory structure: there is an `inspection` folder instead of `test`.

Reply:
Good catch, replaced all incorrect instances.